### ISSUE TITLE
vs-select from vuesax not working and duplicate root node in vuejs

### ISSUE THREAD

**Issue by yvescavalcanti** [lusaxweb/vuesax#340]
I'm working with vuejs 2.5.16, vuesax 3.5.7 and webpack template from vue-cli. When i utilize a vs-select component im my form, the options are not visible, the list size seems to be correct, but the values are all null. And in the Dev Tools the root node gets doubled. Any sugestions?
My enviroment
Ubuntu 16.04
Node 8.12.0
npm 6.2.0
vue-cli 2.9.6 webpack template

`<vs-col vs-w="2">
<label for="sexo">Sexo</label>
<vs-select v-model="pessoa.sexo">
    <vs-select-item v-bind:value="opt.value" 
       v-bind:text="opt.text" v-bind:key="index"
       v-for="(opt, index) in sexo_opt"/>
</vs-select>`

`data(){
    return{
       pessoa.sexo:'M',
       sexo_opt: [{text:'Masculino',value:'M'},{text:'Feminino',value:'F'}]
    }
}`

**Comment by luisDanielRoviraContreras** [comment]
You could do a jsfiddle or codepen with the problem to be able to solve it better, thanks

**Comment by andreyborisenko** [comment]
@luisDanielRoviraContreras Same issue, here is the link https://codesandbox.io/s/03n47p6q6p

**Comment by luisDanielRoviraContreras** [comment]
@andreyborisenko I already found the error is that you do not have the latest version of vuesax please update and tell me

in the latest version 3.6.x, all the properties with the x-xprop were eliminated, now only xprop for a better understanding

**Comment by yvescavalcanti** [comment]
Thanks! I updated to 3.6 and now the vs-selected it works just fine. But the doubled root problem are still. If i remove vs-select, the doubled root become one. I'm changing an old project made with bootstrap vue to vuesax, maybe i'm having a confilct. i'll try to start a clean project.

**Comment by luisDanielRoviraContreras** [comment]
I am going to review because when starting the vs-select component, the items element is moved to the root of the sun, maybe that has some drawback, what, @brunokunace  do you think?

**Comment by Jacrys** [comment]
from my troubleshooting, there are multiple components affected. `vs-select`, `vs-tabs`, and `vs-tooltip` to name a few. These all have one constituent component in common: `transition`.  it seems that this duplicates the node in the virtual DOM (or at least in the scan function of Vue DevTools) and causes duplicate nodes down the length of the tree.  Removing the transition wrapper solves the issue.  I don't know enough about transition to comment really on why this might be happening, though one theory I saw stated it could be due to a transition not knowing when to end  (and therefore not getting removed from the tree), but the transitions I see seem well-formed and compliant with documentation, so I do not know.

**Comment by luisDanielRoviraContreras** [comment]
@yvescavalcanti @Jacrys When they refer to duplicate they refer to the element in the sun or to the virtual sun, as I can verify that it is duplicated to fix it

**Comment by brunokunace** [comment]
> I am going to review because when starting the vs-select component, the items element is moved to the root of the sun, maybe that has some drawback, what, @brunokunace do you think?

i check all component, but i dont repro this, all docs are OK, maybe i initiate a vuecli for test this, importing vuesax on npm.

![image](https://user-images.githubusercontent.com/16597320/47762181-555c8080-dc9a-11e8-9327-72696c67ccec.png)

**Comment by Jacrys** [comment]
My screenshot tool is being difficult and won't screenshot the DevTools, but here is the excerpt:

```
VueReady. Detected Vue 2.5.17.

<Root>
  <Layout> router-view: /components/selects.html
    <Navbar>
      <Page key='__transition-190-vue-component-70-Page'>
        <v-d268a416a8a5e key='__transition-192-vue-component-64-v-d268a416a8a5e'>
          <Box>
          <Box>
            <vuecode>
              <Demos-Select-Default>
          <Box>
            <vuecode>
              <Demos-Select-Color>
          <Box>
            <vuecode>
              <Demos-Select-Autocomplete>
          <Box>
            <vuecode>
              <Demos-Select-Multiple>
          <Box>
            <vuecode>
              <Demos-Select-Labels>
          <Box>
            <vuecode>
              <Demos-Select-Validation>
          <Box>
            <vuecode>
              <Demos-Select-Selectedtext>
          <Box>
            <vuecode>
              <Demos-Select-Group>
        <api>
        <contributors>
        <router-link>
        <router-link>
        <Footer>
      <sidebar-map>
      <Sidebar>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<transition>
<Root>
  <Layout> router-view: /components/selects.html
    <Navbar>
      <Page key='__transition-190-vue-component-70-Page'>
        <v-d268a416a8a5e key='__transition-192-vue-component-64-v-d268a416a8a5e'>
          <Box>
          <Box>
            <vuecode>
              <Demos-Select-Default>
          <Box>
            <vuecode>
              <Demos-Select-Color>
          <Box>
            <vuecode>
              <Demos-Select-Autocomplete>
          <Box>
            <vuecode>
              <Demos-Select-Multiple>
          <Box>
            <vuecode>
              <Demos-Select-Labels>
          <Box>
            <vuecode>
              <Demos-Select-Validation>
          <Box>
            <vuecode>
              <Demos-Select-Selectedtext>
          <Box>
            <vuecode>
              <Demos-Select-Group>
        <api>
        <contributors>
        <router-link>
        <router-link>
        <Footer>
      <sidebar-map>
      <Sidebar>
```
The root node is duplicated after the transitions load.  all attributes are duplicated as well.  This seems to only be in the VDOM inside of Vue, either that or the duplication bug in Vue DevTools hasn't really been fixed.  if it actually is in the VDOM, then it may either be an issue with something in the transitions, a Vue bug, or, as previously mentioned, a bug in the Vue DevTools extension.

If I remove the transition wrapper, the issue no longer duplicates:

```
<Root>
  <Layout> router-view: /components/selects.html
    <Navbar>
      <Page key='__transition-190-vue-component-70-Page'>
        <v-d268a416a8a5e key='__transition-192-vue-component-64-v-d268a416a8a5e'>
          <Box>
          <Box>
            <vuecode>
              <Demos-Select-Default>
          <Box>
            <vuecode>
              <Demos-Select-Color>
          <Box>
            <vuecode>
              <Demos-Select-Autocomplete>
          <Box>
            <vuecode>
              <Demos-Select-Multiple>
          <Box>
            <vuecode>
              <Demos-Select-Labels>
          <Box>
            <vuecode>
              <Demos-Select-Validation>
          <Box>
            <vuecode>
              <Demos-Select-Selectedtext>
          <Box>
            <vuecode>
              <Demos-Select-Group>
        <api>
        <contributors>
        <router-link>
        <router-link>
        <Footer>
      <sidebar-map>
      <Sidebar>
```

**Comment by Jacrys** [comment]
OK, comparing the VDOM models in FF and Chrome with the official release version vs beta Vue DevTools, The problem disappears in the beta version.  Looking through the release notes, this was supposedly fixed in 4.1.5-beta.1, but was apparently regressed in beta.2 or in the release version.  It looks like the Vue DevTools team has fixed it in the latest beta, so let's hope that is put on release soon, and there isn't another feature regression.  

In short, I can confirm this is a bug in Vue DevTools.  There is an existing bug in their scan algorithm that causes this to occur because it bases on the incorrect root (vue-devtools [#645] (https://github.com/vuejs/vue-devtools/issues/645)).

**Comment by Jacrys** [comment]
I'd put money on the fact that @yvescavalcanti and @andreyborisenko are, like me, using Version 4.1.5 of the Vue DevTools:

![image](https://user-images.githubusercontent.com/3784287/47795687-f1b17200-dcf0-11e8-8ebd-b67f4fc14645.png)

**Comment by yvescavalcanti** [comment]
And you are right @Jacrys. I'm using version 4.1.5 beta. So i can close the case. Let's wait for a new Dev Tools update. Thanks to everybody

**Comment by isferrei** [comment]
@luisDanielRoviraContreras Hi Luis, I'm getting the same issue, however I can't update the version of Vuesax in the whole project cause it's changing some components appearence. Could you please help me to solve it ?
